app/uk/gov/hmrc/gform/sharedmodel/formtemplate/AuthConfig.scala

```diff
@@ -118,34 +118,49 @@ case class HmrcAgentModule(agentAccess: AgentAccess) extends AuthConfig
 case class HmrcAgentWithEnrolmentModule(agentAccess: AgentAccess, enrolmentAuth: EnrolmentAuth) extends AuthConfig
 
 object HasEnrolmentSection {
-  def unapply(ac: AuthConfig): Option[(ServiceId, EnrolmentSection, EnrolmentPostCheck)] = ac match {
-    case HmrcEnrolmentModule(EnrolmentAuth(serviceId, DoCheck(_, RequireEnrolment(es), check))) =>
-      Some((serviceId, es, check))
-    case HmrcAgentWithEnrolmentModule(_, EnrolmentAuth(serviceId, DoCheck(_, RequireEnrolment(es), check))) =>
-      Some((serviceId, es, check))
-    case _ => None
-  }
+  def unapply(ac: AuthConfig): Option[(ServiceId, EnrolmentSection, EnrolmentPostCheck, EnrolmentAction)] =
+    ac match {
+      case HmrcEnrolmentModule(EnrolmentAuth(serviceId, DoCheck(_, RequireEnrolment(es, enrolmentAction), check))) =>
+        Some((serviceId, es, check, enrolmentAction))
+      case HmrcAgentWithEnrolmentModule(
+          _,
+          EnrolmentAuth(serviceId, DoCheck(_, RequireEnrolment(es, enrolmentAction), check))) =>
+        Some((serviceId, es, check, enrolmentAction))
+      case _ => None
+    }
 }
 
 object AuthConfig {
 
   def toEnrolmentPostCheck(maybeRegimeId: Option[RegimeId]): EnrolmentPostCheck =
     maybeRegimeId.fold(NoCheck: EnrolmentPostCheck)(RegimeIdCheck.apply)
 
+  def enrolmentActionMatch(enrolmentAction: Option[EnrolmentAction]): EnrolmentAction = enrolmentAction match {
```

Comment:
Body of this method may be shortened to just `enrolmentAction.getOrElse(NoAction)`